module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -604,13 +606,22 @@ public function getHolding($bibId, array $patron = null, array $options = [])
                     $item->itemLevelCallNumberPrefix ?? '',
                     $item->itemLevelCallNumber ?? ''
                 );
+
+                $dueDate = new DateTime(
```

Comment:
Should we do this conditionally, only when $item->status->date is set? (I assume many items will not have a due date, if they are not checked out).

Reply:
I had found a date on every item that I had checked and thus made an assumption it would always be there

`"status": {
                "name": "Available",
                "date": "2021-06-25T15:47:44.956+00:00"
            },`

Thanks to your question, I found that while this is true, there are other status messages that also returned a date and that will cause a problem:

`"status": {
                "name": "In process",
                "date": "2022-06-21T10:32:07.860+00:00"
            }`

I'll investigate this further, as I think it is less a function of whether the date is set, but whether the status is one for which we would want to show a date. Thanks for pointing this out, I had looked only at available and unavailable items during testing and it looked good - but I found a "in process" item that then showed a due date, that was obviously wrong

Reply:
Thanks for the progress on this, @cwolfebsco. Just one suggestion: you might consider changing the assignment to $dueDateValue to read:

```
                    $dueDateValue = $this->dateConverter
                        ->convertToDisplayDateAndTime('U', $dueDate->format('U'));
```

This way, instead of a hard-coded display date format in the driver, this will respect the displayDateFormat/displayTimeFormat settings in config.ini. What do you think?

Reply:
Makes perfect sense :) I'll submit a change...

Reply:
BTW I just found a fallacy in this implementation during more testing :( - this needs more than just a small change, I am afraid

Reply:
Sorry to hear that; let me know if I can be of any help with testing or general brainstorming.

Reply:
Thanks for the offer, I think I need some brain storming here in a moment.

Basically I made a massive mistake as to where to get the dueDate from. In the code you will notice that I was trying to get it from $item->status->date and I had a screenshot attached, but it clearly shows 02 July 2021. I am obviously living in the past.

As far as I can tell the correct way is to call /circulation/loans with the itemId and get the dueDate from there.

I do however want to ask about the Universal date format:

For this time String:
2022-12-15T23:59:00.000+00:00. (<== UTC)

This is the output in my environment
15-12-2022 23:59 (<== currently London is BST = UTC+1)

Unless you have short term loans in your library, is the time truly desired? If so, would it make sense to let users configure the desired format in folio.ini? 
<img width="801" alt="Screenshot 2022-07-05 at 19 04 27" src="https://user-images.githubusercontent.com/62254772/177379892-cf4b3a55-0fe7-4c1d-8868-ab0394f505a2.png">